packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -468,414 +425,314 @@ class _MinuteControl extends StatelessWidget {
-        ),
-        piece(
-          fragment1: dayPeriod(),
-        ),
-      );
-    case TimeOfDayFormat.H_colon_mm:
-      return format(piece(
-        pivotIndex: 1,
-        fragment1: hour(),
-        fragment2: string(_TimePickerHeaderId.colon, ':'),
-        fragment3: minute(),
-      ));
-    case TimeOfDayFormat.HH_dot_mm:
-      return format(piece(
-        pivotIndex: 1,
-        fragment1: hour(),
-        fragment2: string(_TimePickerHeaderId.dot, '.'),
-        fragment3: minute(),
-      ));
-    case TimeOfDayFormat.a_space_h_colon_mm:
-      return format(
-        piece(
-          fragment1: dayPeriod(),
-        ),
-        piece(
-          pivotIndex: 1,
-          fragment1: hour(),
-          fragment2: string(_TimePickerHeaderId.colon, ':'),
-          fragment3: minute(),
-        ),
-      );
-    case TimeOfDayFormat.frenchCanadian:
-      return format(piece(
-        pivotIndex: 1,
-        fragment1: hour(),
-        fragment2: string(_TimePickerHeaderId.hString, 'h'),
-        fragment3: minute(),
-      ));
-    case TimeOfDayFormat.HH_colon_mm:
-      return format(piece(
-        pivotIndex: 1,
-        fragment1: hour(),
-        fragment2: string(_TimePickerHeaderId.colon, ':'),
-        fragment3: minute(),
-      ));
-  }
-
-  return null;
-}
+  @override
+  Widget build(BuildContext context) {
+    final MaterialLocalizations materialLocalizations = MaterialLocalizations.of(context);
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final TimePickerThemeData timePickerTheme = TimePickerTheme.of(context);
+    final bool isDark = colorScheme.brightness == Brightness.dark;
+    final Color selectedTextColor = timePickerTheme.dayPeriodSelectedTextColor ?? colorScheme.primary;
+    final Color unselectedTextColor = timePickerTheme.dayPeriodUnselectedTextColor ?? colorScheme.onSurface.withOpacity(0.60);
+    final Color selectedColor = timePickerTheme.dayPeriodSelectedColor ?? colorScheme.primary.withOpacity(isDark ? 0.24 : 0.12);
+    // The unselected day period should match the overall picker dialog color.
+    // Making it transparent enables that without being redundant and allows
+    // the optional elevation overlay for dark mode to be visible.
+    final Color unselectedColor = timePickerTheme.dayPeriodUnselectedColor ?? Colors.transparent;
 
-class _TimePickerHeaderLayout extends MultiChildLayoutDelegate {
-  _TimePickerHeaderLayout(this.orientation, this.format)
-    : assert(orientation != null),
-      assert(format != null);
+    final bool amSelected = selectedTime.period == DayPeriod.am;
+    final TextStyle textStyle = TimePickerTheme.of(context).dayPeriodTextStyle ?? Theme.of(context).textTheme.subtitle1;
+    final TextStyle amStyle = textStyle.copyWith(
+      color: amSelected ? selectedTextColor: unselectedTextColor,
+    );
+    final TextStyle pmStyle = textStyle.copyWith(
+      color: !amSelected ? selectedTextColor: unselectedTextColor,
```

Comment:
This implies that if amSelected is false, then 'pmSelected' would be true, consider making `pmSelected => !amSelected`

Reply:
done